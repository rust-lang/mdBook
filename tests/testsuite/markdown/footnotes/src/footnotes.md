# Footnote tests

Footnote example[^1], or with a word[^word].

[^1]: This is a footnote.

[^word]: A longer footnote.
    With multiple lines. [Link to other](other.md).
    With a reference inside.[^1]

There are multiple references to word[^word].

Footnote without a paragraph[^para]

[^para]:
    1. Item one
       1. Sub-item
    2. Item two

Footnote with multiple paragraphs[^multiple]

[^define-before-use]: This is defined before it is referred to.

[^multiple]:
    One

    Two

    Three

[^unused]: This footnote is defined by not used.

Footnote name with wacky characters[^"wacky"]

[^"wacky"]: Testing footnote id with special characters.

Testing when referring to something earlier.[^define-before-use]

Footnote that is defined multiple times.[^multiple-definitions]

[^multiple-definitions]: This is the first definition of the footnote with tag multiple-definitions

And another[^in-between] that references the duplicate again.[^multiple-definitions]

[^in-between]: Footnote between duplicates.

[^multiple-definitions]: This is the second definition of the footnote with tag multiple-definitions

Multiple footnotes in a row.[^a][^b][^c]

[^a]: Footnote 1
[^b]: Footnote 2
[^c]: Footnote 3
