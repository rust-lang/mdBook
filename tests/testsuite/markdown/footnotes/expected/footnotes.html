<h1 id="footnote-tests"><a class="header" href="#footnote-tests">Footnote tests</a></h1>
<p>Footnote example<sup class="footnote-reference" id="fr-1-1"><a href="#footnote-1">1</a></sup>, or with a word<sup class="footnote-reference" id="fr-word-1"><a href="#footnote-word">2</a></sup>.</p>
<p>There are multiple references to word<sup class="footnote-reference" id="fr-word-2"><a href="#footnote-word">2</a></sup>.</p>
<p>Footnote without a paragraph<sup class="footnote-reference" id="fr-para-1"><a href="#footnote-para">3</a></sup></p>
<p>Footnote with multiple paragraphs<sup class="footnote-reference" id="fr-multiple-1"><a href="#footnote-multiple">4</a></sup></p>
<p>Footnote name with wacky characters<sup class="footnote-reference" id="fr-&quot;wacky&quot;-1"><a href="#footnote-&quot;wacky&quot;">5</a></sup></p>
<p>Testing when referring to something earlier.<sup class="footnote-reference" id="fr-define-before-use-1"><a href="#footnote-define-before-use">6</a></sup></p>
<p>Footnote that is defined multiple times.<sup class="footnote-reference" id="fr-multiple-definitions-1"><a href="#footnote-multiple-definitions">7</a></sup></p>
<p>And another<sup class="footnote-reference" id="fr-in-between-1"><a href="#footnote-in-between">8</a></sup> that references the duplicate again.<sup class="footnote-reference" id="fr-multiple-definitions-2"><a href="#footnote-multiple-definitions">7</a></sup></p>
<p>Multiple footnotes in a row.<sup class="footnote-reference" id="fr-a-1"><a href="#footnote-a">9</a></sup> <sup class="footnote-reference" id="fr-b-1"><a href="#footnote-b">10</a></sup> <sup class="footnote-reference" id="fr-c-1"><a href="#footnote-c">11</a></sup></p>
<hr>
<ol class="footnote-definition"><li id="footnote-1">
<p>This is a footnote. <a href="#fr-1-1">↩</a> <a href="#fr-1-2">↩2</a></p>
</li>
<li id="footnote-word">
<p>A longer footnote.
With multiple lines. <a href="other.html">Link to other</a>.
With a reference inside.<sup class="footnote-reference" id="fr-1-2"><a href="#footnote-1">1</a></sup> <a href="#fr-word-1">↩</a> <a href="#fr-word-2">↩2</a></p>
</li>
<li id="footnote-para">
<ol>
<li>Item one
<ol>
<li>Sub-item</li>
</ol>
</li>
<li>Item two</li>
</ol>
 <a href="#fr-para-1">↩</a></li>
<li id="footnote-multiple">
<p>One</p>
<p>Two</p>
<p>Three <a href="#fr-multiple-1">↩</a></p>
</li>
<li id="footnote-&quot;wacky&quot;">
<p>Testing footnote id with special characters. <a href="#fr-&quot;wacky&quot;-1">↩</a></p>
</li>
<li id="footnote-define-before-use">
<p>This is defined before it is referred to. <a href="#fr-define-before-use-1">↩</a></p>
</li>
<li id="footnote-multiple-definitions">
<p>This is the first definition of the footnote with tag multiple-definitions <a href="#fr-multiple-definitions-1">↩</a> <a href="#fr-multiple-definitions-2">↩2</a></p>
</li>
<li id="footnote-in-between">
<p>Footnote between duplicates. <a href="#fr-in-between-1">↩</a></p>
</li>
<li id="footnote-a">
<p>Footnote 1 <a href="#fr-a-1">↩</a></p>
</li>
<li id="footnote-b">
<p>Footnote 2 <a href="#fr-b-1">↩</a></p>
</li>
<li id="footnote-c">
<p>Footnote 3 <a href="#fr-c-1">↩</a></p>
</li>
</ol>